refract-spec.md

```diff
@@ -12,89 +12,286 @@ This document conforms to RFC 2119, which says:
+
+##### Members
+
+- (string)
+- (String Element)
+
+#### Number Type (enum)
+
+A Number Type in Refract can be either a number value or a Number Element.
+
+##### Members
+
+- (number)
+- (Number Element)
+
+#### Boolean Type (enum)
+
+A Boolean Type in Refract can be either a boolean value or a Boolean Element.
+
+##### Members
+
+- (boolean)
+- (Boolean Element)
+
+### Structure Types
+
+Structure Types provide structure for Refract documents. These types consist of:
+
+* Array Type
+* Object Type
+* Member Type
+
+#### Array Type (array)
+
+An Array Type MAY be an array of all Refract primitives except the Object Type. This ensures there is no confusion between an object and an actual Refract Object Type.
+
+##### Members
+
+- (enum)
+    - (Null Type)
+    - (String Type)
+    - (Number Type)
+    - (Boolean Type)
+    - (Array Type)
+    - (Element)
+
+##### Example
+
+This example shows an array containing a sample of each primitive type along with an element.
+
+```json
+[
+  null,
+  "foo",
+  100,
+  true,
+  [1, 2, 3],
+  {
+    "element": "string",
+    "content": "bar"
+  }
+]
+```
+
+#### Object Type (enum)
+
+Object Types provide a key/value structure for Refract.
+
+When converting an Object Type to a primitive JSON type, and the Object Type is a JSON object, no processing is needed. If the Object Type is an array, the following rules apply:
+
+1. If the item of the array is a member type, the `key` MUST be the key, and the `value` MUST be the value.
+1. If the item of the array is a Refract Element, they element name MUST be the key, and the element content MUST be the value.
+
+##### Members
+
+- (object)
+- (array)
+    - (enum)
+        - (Member Type)
         - (Element)
 
-- `attributes` (enum)
+##### Examples
 
-    The `attributes` property defines attributes about the given instance of the element, as specified by the `element` property. It MAY be an array of elements, or it MAY be an object. The keys of the object SHOULD be defined element types with valid values for the corresponding element types.
+An Object Type that is an object will look like this example.
 
-    - Members
-        - (object)
-        - array[Element]
+```json
+{
+  "foo": "bar",
+  "hello": "world"
+}
+```
 
-- `content` (enum)
+An Object Type that is an array of Member Types, will look like this. If this structure were to be converting back to a JSON object, it would look like the previous example.
 
-    The `content` property defines the content of the instance of the specified element type. The value MAY be any of the JSON primitive types, an element, or an array of elements.
+```json
+[
+  {
+    "key": "foo",
+    "value": "bar"
+  },
+  {
+    "key": "hello",
+    "value": "world"
+  }
+]
+```
 
-    - Members
-        - (null)
+When a `foo` element is defined that has a string for its content, the element MAY be used instead of a Member Type as an item in the Object Type array.
+
+```json
+[
+  {
+    "element": "foo",
```

Comment:
@smizell I fail to see the value add for this definition as noted above: https://github.com/refractproject/refract-spec/pull/5/files#diff-4eef42b68375cd377e0657ef7ea34f6bR113

It convolutes the idea of an element type with a key of an object and I don't see the benefit for the added complexity. It's like which of these elements is not like any of the others? 

I think a much more declarative definition would be:

If the item of the array is an array, then the array must contain a key and a value element.

```json
[
  [['key', {}, {}, 'foo'], ['value', {}, {}, 'bar'],
  ['key', {}, {}, 'hello'], ['value', {}, {}, 'world']]
]
```
This is far more declarative makes clear this thing is an object type, and I also think in MSON, based on the idea of variable keys, etc, having metadata on a key is going to be useful, particularly with complex objects, potentially. Adding some metadata to ['foo', {}, {}, 'bar'] and that actually means an object member is limiting in this case.

Further, if I ran across:

```json
['array', {}, {}, [[['foo', {}, {}, 'bar']]]
```
is that content an array of foo element or an object.

I think this is a complete stretch and overloads the definition of an element type in a way that is unique only to this case and therefore IMO, not valuable. 

So, back to the question, what is the value add of this vs. the cost of complexity and confusion?